production/helm/loki/templates/_helpers.tpl

```diff
@@ -359,3 +359,14 @@ Create the service endpoint including port for MinIO.
 {{- define "enterprise-logs.canarySecret" }}
 {{- .Values.enterprise.canarySecret | default (printf "%s-canary-secret" (include "loki.name" . )) -}}
 {{- end -}}
+
+{{/* Configure enableServiceLinks in pod */}}
+{{- define "loki.enableServiceLinks" -}}
+{{- if semverCompare ">=1.13-0" .Capabilities.KubeVersion.GitVersion -}}
+{{- if or (.Values.loki.enableServiceLinks) (eq (.Values.loki.enableServiceLinks | toString) "<nil>") -}}
+enableServiceLinks: true
+{{- else -}}
+enableServiceLinks: false
+{{- end -}}
```

Comment:
I still don't love the to string and comparison to the string "<nil>", as I feel like that's too tightly coupled to go template implementation details. How would you feel about switching the logic to avoid that?

```suggestion
{{- if not .Values.loki.enableServiceLinks -}}
enableServiceLinks: false
{{- else -}}
enableServiceLinks: true
{{- end -}}
```

Reply:
this result in a `true` if it's explicitly `true` or omitted

Reply:
This won't work because if the value is omitted, it is treated as false

Reply:
I made the change to `{{- if or (.Values.loki.enableServiceLinks) (ne .Values.loki.enableServiceLinks false) -}}` so instead of using `toString` will check if it is false, this way will decouple with go templating implementation, hope this is ok for you

Reply:
I'm not following. My understanding was we want `enableServiceLinks` to be `true` if it's either explicitly set to `true` or if it's omitted. The above should work since we set the default to `true`, so if omitted it should use the default value. I checked out your branch, and made only the change above, and these are my results.

`helm template . --set loki.enableServiceLinks=false --show-only templates/gateway/deployment-gateway.yaml`:
```
---
# Source: loki/templates/gateway/deployment-gateway.yaml
...
    spec:
      serviceAccountName: loki
      enableServiceLinks: false
...
```

`helm template . --set loki.enableServiceLinks=true --show-only templates/gateway/deployment-gateway.yaml`:
```
---
# Source: loki/templates/gateway/deployment-gateway.yaml
...
    spec:
      serviceAccountName: loki
      enableServiceLinks: true
...
```

`helm template . --show-only templates/gateway/deployment-gateway.yaml`
```
---
# Source: loki/templates/gateway/deployment-gateway.yaml
...
    spec:
      serviceAccountName: loki
      enableServiceLinks: true
...
```

Sorry for the confusion, but is this not the correct behavior?

Reply:
This second condition is to cover the case where the "enableServiceLinks" is missing in the `values.yaml`

Reply:
If it's missing in `values.yaml` I don't understand why it wouldn't get the default value?